Failure on aarch64-linux [(full log)](https://logs.nix.gsc.io/?key=nixos/nixpkgs.34676&attempt_id=80ffa2af-c23e-4720-8a6c-59fdecdfa887)

<details><summary>Partial log (click to expand)</summary><p>

```
g++: error: qrc_resources.cpp: No such file or directory
DQT_NO_DEBUG -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB --include ./moc_predefs.h -I/nix/store/5v3axjsspmn16kiknrhp9cypbnngw0pl-qtbase-5.10.0-dev/mkspecs/linux-g++ -I/build/diffpdf-2.1.3 -I/nix/store/38jpy017q88q6fpj642dhkb7a10dv8j7-poppler-qt5-0.56.0-dev/include/poppler/cpp -I/nix/store/38jpy017q88q6fpj642dhkb7a10dv8j7-poppler-qt5-0.56.0-dev/include/poppler/qt5 -I/nix/store/5v3axjsspmn16kiknrhp9cypbnngw0pl-qtbase-5.10.0-dev/include/QtPrintSupport -I/nix/store/5v3axjsspmn16kiknrhp9cypbnngw0pl-qtbase-5.10.0-dev/include/QtWidgets -I/nix/store/5v3axjsspmn16kiknrhp9cypbnngw0pl-qtbase-5.10.0-dev/include/QtGui -I/nix/store/5v3axjsspmn16kiknrhp9cypbnngw0pl-qtbase-5.10.0-dev/include/QtCore -I/nix/store/5v3axjsspmn16kiknrhp9cypbnngw0pl-qtbase-5.10.0-dev/include -I/nix/store/h3n7anzrbr8s8909jpr6sdqmcgbm4y7g-libxml2-2.9.7-dev/include -I/nix/store/f7j8af7gdwsy0gg8xchkw77n97d2pxn4-zlib-1.2.11-dev/include -I/nix/store/58lj4z7cz5zys9fivfacycv9fhmiclcn-libxslt-1.1.32-dev/include -I/nix/store/y6am5ddw0cdlid35hf116hc7bim3b57f-openssl-1.0.2n-dev/include -I/nix/store/wq7k5dws65yxmlhd8i6wmsjxv04l26kz-sqlite-3.21.0-dev/include -I/nix/store/y23yqdvfcqgvv57gr3785chqymplswb8-harfbuzz-icu-1.7.1-dev/include -I/nix/store/i6qnahylmlclr0bxyqiq3n88j0kjj0y0-graphite2-1.3.6/include -I/nix/store/r6i0fdja2g5c6ydsm2ag4q06iwn5wc5z-icu4c-59.1-dev/include -I/nix/store/7xw1798p6xh2bankf7x1imxznf47dncm-harfbuzz-1.7.1-dev/include -I/nix/store/c7xcmpq3vrx1gv3rawbfs8w26gglc75g-libjpeg-turbo-1.5.3-dev/include -I/nix/store/28hv0yafp3xix0pisj5iix0lm1fmr0i4-libpng-apng-1.6.34-dev/include -I/nix/store/w4hkcdkn4px20digygfmfmy4r4s7xf5g-libtiff-4.0.9-dev/include -I/nix/store/b82qjbn6578i31jws7x62cn1hz1hx7qm-xz-5.2.3-dev/include -I/nix/store/45240vpmkgpabm1i8z63anr0dgmi659f-pcre2-10.23-dev/include -I/nix/store/23zw0hqnn863pafzl0ppf2yfh8kkhg66-dbus-1.10.24-dev/include -I/nix/store/7kfdq337p0hpjpjkp5pq96b7dm3mh172-expat-2.2.5-dev/include -I/nix/store/5hdjm7ddqzp34s2sdlvglhmra6h4prhj-glib-2.54.3-dev/include -I/nix/store/1mhfdgg6yb40fh0dvhlrjlb69yh8hpr0-libffi-3.2.1-dev/include -I/nix/store/j44jdinzg369qr7b3x89nazl65zwywjf-glibc-iconv-2.26-131/include -I/nix/store/mb6dbjzk0bgp0b7dzpyk8y1gjsazx7p1-systemd-234-dev/include -I/nix/store/7r5ny3pz1xhkrhnrbd040i1gqkqgkxpk-fontconfig-2.12.1-dev/include -I/nix/store/83a5ziz460cawi55hx846h7m84qc7bms-freetype-2.7.1-dev/include -I/nix/store/j97n4pfwq1szmzwscxwgvjhfp77rsgj5-bzip2-1.0.6.0.1-dev/include -I/nix/store/s3j31s8as99sjfmka61qmkg0f4zzkkq9-libX11-1.6.5-dev/include -I/nix/store/0m2pxx7c33yk6kdyq3nmj596bdy1wkji-libxcb-1.12-dev/include -I/nix/store/09iciblshchw1gb4jwld7mji0gz3a1l1-xproto-7.0.31/include -I/nix/store/5gn8rp0ibzh4b504g5s0711mi5g82awk-kbproto-1.0.7/include -I/nix/store/s7vxj8blrbdfk5xs5hy118zjk5wvypar-libXcomposite-0.4.4-dev/include -I/nix/store/dr43fz3lzh67wh29hf8367c0xy9mf5zc-libXfixes-5.0.2-dev/include -I/nix/store/dma8m4v4kklvv46r98b6v06zh6yjd48r-fixesproto-5.0/include -I/nix/store/2kbrxg40vd2z8dca7aafwfik4fhg79wn-xextproto-7.3.0/include -I/nix/store/bpj3xq5wv3iiakqlsflh7mzvks1acqmb-compositeproto-0.4.2/include -I/nix/store/mrsqca1cbxb7y75fyppa5yrydjhbjn3f-libXext-1.3.3-dev/include -I/nix/store/8bbl1ff94q72dlp36jaqc6n3jsf5kn5b-libXau-1.0.8-dev/include -I/nix/store/bhh6zr6284ypk52gsj156j0qffd12s2f-libXi-1.7.9-dev/include -I/nix/store/3wrndfqc8xv4f3ac8ryb11d2dm4kwffg-inputproto-2.3.2/include -I/nix/store/x3i1gi1lvs2qax5380jci6qcn9bhd9kv-libXrender-0.9.10-dev/include -I/nix/store/vzwxfd8icdpcqhvpg8whlhxxszhaw1ww-renderproto-0.11.1/include -I/nix/store/5b8bwz8plgzcl3b71kx6h49gcd83wgpc-libxkbcommon-0.7.2-dev/include -I/nix/store/6x9bjnl7q934kq2kcm0p0hms2nz607pm-xcb-util-0.4.0-dev/include -I/nix/store/4g5mvn5s29wivcsm7hkfh8jdcby7kpv8-xcb-util-image-0.4.0-dev/include -I/nix/store/3fgsijj0d6271k6gvxb3vfsg298i13l3-xcb-util-keysyms-0.4.0-dev/include -I/nix/store/llvrmd8qp2nzk84yys2r8gplac2697yz-xcb-util-renderutil-0.3.9-dev/include -I/nix/store/y0vkywrnwwwhqiz24x08kn1fnmpvq81p-xcb-util-wm-0.4.1-dev/include -I/nix/store/75dwi1zzfnwcgll7ymphgmjiq5c0dmx5-mesa-noglu-17.3.3-dev/include -I/nix/store/3zfcvrsszvwfxf0may42idmb8imgj1rk-libXdamage-1.1.4-dev/include -I/nix/store/d7hpldg1gcv5niifi0q2h8p5jzal700r-damageproto-1.2.1/include -I/nix/store/i1nbdl51i7rygl3h3iy2ha2pyjyg5fi8-libXxf86vm-1.1.4-dev/include -I/nix/store/i212877nfp9gqjsqablff54ww4alzikv-xf86vidmodeproto-2.3.1/include -I/nix/store/v3z9fywljw891w4nr16rgzf3hmc0448n-libdrm-2.4.89-dev/include -I/nix/store/fzhzdav67q94d88yg63f7crqq158gh6f-qttools-5.10.0-dev/include -I/nix/store/38jpy017q88q6fpj642dhkb7a10dv8j7-poppler-qt5-0.56.0-dev/include -I/nix/store/3jx564569g8vq2slx0nk7828k1lrq7b8-openjpeg-2.3.0-dev/include -I/nix/store/nv699vkn549j95sk6vrxafv8lakx7rh5-lcms2-2.8-dev/include -I/nix/store/prhrmz1g5gy71bp6fm6c1gdn2fvb8p02-cairo-1.14.10-dev/include -I/nix/store/m7r485y7n3w31znmd96pibbymi1l9rds-pixman-0.34.0/include -I/nix/store/bsxpyvki3sc3z3d57s04zh00sim0basf-curl-7.58.0-dev/include -I/nix/store/m4y23ikdklm6501zx28vs8k4aw9n8pqq-nghttp2-1.24.0-dev/include -I/nix/store/13ick0ml8yz5z54y0lw7qkl3k2qkv2ga-libkrb5-1.15.2-dev/include -I/nix/store/4jlbdjwc20g3aqxflshszimqs7hmrqq9-libssh2-1.8.0-dev/include -I/nix/store/xzd7gljmi0xhiqwyah9z1d8dl2r9m2im-gcc-6.4.0/include/c++/6.4.0 -I/nix/store/xzd7gljmi0xhiqwyah9z1d8dl2r9m2im-gcc-6.4.0/include/c++/6.4.0/aarch64-unknown-linux-gnu -I/nix/store/xzd7gljmi0xhiqwyah9z1d8dl2r9m2im-gcc-6.4.0/include/c++/6.4.0/backward -I/nix/store/xzd7gljmi0xhiqwyah9z1d8dl2r9m2im-gcc-6.4.0/lib/gcc/aarch64-unknown-linux-gnu/6.4.0/include -I/nix/store/xzd7gljmi0xhiqwyah9z1d8dl2r9m2im-gcc-6.4.0/include -I/nix/store/xzd7gljmi0xhiqwyah9z1d8dl2r9m2im-gcc-6.4.0/lib/gcc/aarch64-unknown-linux-gnu/6.4.0/include-fixed -I/nix/store/ma9nn33d5xvqqsg4vcg58yrl0ngmqnbh-glibc-2.26-131-dev/include mainwindow.hpp -o moc_mainwindow.cpp
g++: fatal error: no input files
compilation terminated.
make: *** [Makefile:518: qrc_resources.o] Error 1
make: *** Waiting for unfinished jobs....
/nix/store/7fxbh1yhagvwbdrmdyyy5ghcjhwjndhs-bash-4.4-p12/bin/bash: DQT_NO_DEBUG: command not found
make: [Makefile:426: moc_mainwindow.cpp] Error 127 (ignored)
builder for '/nix/store/wl4wma6m9k19zmcssl26q0clngxh8gc0-diffpdf-2.1.3.drv' failed with exit code 2
error: build of '/nix/store/wl4wma6m9k19zmcssl26q0clngxh8gc0-diffpdf-2.1.3.drv' failed
```
</p></details>

--- a/pkgs/applications/misc/diffpdf/default.nix
+++ b/pkgs/applications/misc/diffpdf/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, qt4, poppler_qt4, qmake4Hook }:
+{ stdenv, fetchurl, fetchpatch, qmake, qttools, qtbase, poppler_qt5 }:
 
 stdenv.mkDerivation rec {
   version = "2.1.3";
@@ -9,13 +9,19 @@ stdenv.mkDerivation rec {
     sha256 = "0cr468fi0d512jjj23r5flfzx957vibc9c25gwwhi0d773h2w566";
   };
 
-  patches = [ ./fix_path_poppler_qt4.patch ];
+  patches = [
+    (fetchpatch {
+      url = https://raw.githubusercontent.com/gentoo/gentoo/9b971631588ff46e7c2d501bc35cd0d9ce2d98e2/app-text/diffpdf/files/diffpdf-2.1.3-qt5.patch;
+      sha256 = "0sax8gcqcmzf74hmdr3rarqs4nsxmml9qmh6pqyjmgl3lypxhafg";
+    })
+    ./fix_path_poppler_qt5.patch
+  ];
 
-  buildInputs = [ qt4 poppler_qt4 ];
-  nativeBuildInputs = [ qmake4Hook ];
+  nativeBuildInputs = [ qmake qttools ];
+  buildInputs = [ qtbase poppler_qt5 ];
 
   preConfigure = ''
-    substituteInPlace diffpdf.pro --replace @@NIX_POPPLER_QT4@@ ${poppler_qt4.dev}
+    substituteInPlace diffpdf.pro --replace @@NIX_POPPLER_QT5@@ ${poppler_qt5.dev}
     lrelease diffpdf.pro
   '';
 

--- a/pkgs/applications/misc/diffpdf/fix_path_poppler_qt5.patch
+++ b/pkgs/applications/misc/diffpdf/fix_path_poppler_qt5.patch
@@ -2,15 +2,15 @@ diff -uNr diffpdf-2.1.3/diffpdf.pro diffpdf-2.1.3-new/diffpdf.pro
 --- diffpdf-2.1.3/diffpdf.pro	2013-10-15 09:01:22.000000000 +0200
 +++ diffpdf-2.1.3-new/diffpdf.pro	2015-07-07 23:13:36.445572148 +0200
 @@ -47,9 +47,9 @@
- 	INCLUDEPATH += /c/poppler_lib/include/poppler/qt4
+ 	INCLUDEPATH += /c/poppler_lib/include/poppler/qt5
  	LIBS += -Wl,-rpath -Wl,/c/poppler_lib/bin -Wl,-L/c/poppler_lib/bin
      } else {
--	exists(/usr/include/poppler/qt4) {
+-	exists(/usr/include/poppler/qt5) {
 -	    INCLUDEPATH += /usr/include/poppler/cpp
--	    INCLUDEPATH += /usr/include/poppler/qt4
-+	exists(@@NIX_POPPLER_QT4@@/include/poppler/qt4) {
-+	    INCLUDEPATH += @@NIX_POPPLER_QT4@@/include/poppler/cpp
-+	    INCLUDEPATH += @@NIX_POPPLER_QT4@@/include/poppler/qt4
+-	    INCLUDEPATH += /usr/include/poppler/qt5
++	exists(@@NIX_POPPLER_QT5@@/include/poppler/qt5) {
++	    INCLUDEPATH += @@NIX_POPPLER_QT5@@/include/poppler/cpp
++	    INCLUDEPATH += @@NIX_POPPLER_QT5@@/include/poppler/qt5
  	} else {
  	    INCLUDEPATH += /usr/local/include/poppler/cpp
- 	    INCLUDEPATH += /usr/local/include/poppler/qt4
+ 	    INCLUDEPATH += /usr/local/include/poppler/qt5

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -16442,7 +16442,7 @@ with pkgs;
 
   mupdf = callPackage ../applications/misc/mupdf { };
 
-  diffpdf = callPackage ../applications/misc/diffpdf { };
+  diffpdf = libsForQt5.callPackage ../applications/misc/diffpdf { };
 
   mlocate = callPackage ../tools/misc/mlocate { };
 

